templates/php/src/Query.php.twig

```diff
@@ -2,112 +2,145 @@
 
 namespace {{ spec.title | caseUcfirst }};
 
-class Query
+class Query implements \JsonSerializable
 {
+    protected string $method;
+    protected ?string $attribute;
+    protected ?array $values;
+
+    public function __construct(string $method, ?string $attribute = null, $values = null)
+    {
+        $this->method = $method;
+        $this->attribute = $attribute;
+        
+        if (is_null($values) || is_array($values)) {
+            $this->values = $values;
+        } else {
+            $this->values = [$values];
+        }
+
+    }
+
+    public function __toString(): string
+    {
+        return json_encode($this->jsonSerialize());
+    }
+
+    public function jsonSerialize()
+    {
+        return array_filter([
+            'method' => $this->method,
+            'attribute' => $this->attribute,
+            'values' => $this->values,
+        ], function ($value) {
+            return !is_null($value);
+        });
```

Comment:
You can omit the callback here, the default is to remove null values

Reply:
`json_encode` does not remove the null values. Are you referring to something else? I think it's safer to keep this

Reply:
From the `array_filter`, we can just have it like:

```suggestion
        ]);
```